lib/logster/group.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+module Logster
+  class Group
+    MAX_SIZE = 100
+
+    attr_reader :key, :messages_keys, :timestamp, :messages
+    attr_accessor :changed
+
+    def initialize(key, messages_keys = [], timestamp: nil)
+      @key = key
+      @messages_keys = messages_keys || []
+      @timestamp = timestamp || 0
+      @changed = true
+    end
+
+    def self.from_json(json)
+      hash = JSON.parse(json)
+      group = new(
+        hash["key"],
+        hash["messages_keys"],
+        timestamp: hash["timestamp"]
+      )
+      group.changed = false
+      group
+    end
+
+    def self.max_size
+      (defined?(@max_size) && @max_size) || MAX_SIZE
+    end
+
+    def self.max_size=(num) # used in tests
+      @max_size = num
+    end
```

Comment:
I don't have a very strong opinion here, but I would like to avoid setting "production" code to have the nice syntax in the test.
Maybe, we can remove that method and in test use uglier https://apidock.com/ruby/Module/class_variable_set

Reply:
I've made this change but I went with instance variables for the class rather than class variables because class variables are shared with instances of the class and I don't think instances need to have access to it.